spec/queries/get_change_offer_options_spec.rb

```diff
@@ -0,0 +1,126 @@
+require 'rails_helper'
+
+RSpec.describe GetChangeOfferOptions do
+  include CourseOptionHelpers
+
+  let(:ratifying_provider) { create(:provider) }
+  let(:self_ratified_course) { create(:course, :open_on_apply, provider: ratifying_provider) }
+  let(:externally_ratified_course) { create(:course, :open_on_apply, accredited_provider: ratifying_provider) }
+  let(:provider_user) { create(:provider_user) }
+
+  def service(provider_user, course)
+    GetChangeOfferOptions.new(user: provider_user, current_course: course)
+  end
+
+  def set_user_permission(provider, make_decisions)
+    provider_user.provider_permissions << ProviderPermissions.new(
+      provider: provider,
+      make_decisions: make_decisions,
+    )
+  end
+
+  def set_provider_permissions(training_provider, make_d1, ratifying_provider, make_d2)
+    create(
+      :provider_relationship_permissions,
+      training_provider: training_provider,
+      ratifying_provider: ratifying_provider,
+      training_provider_can_make_decisions: make_d1,
+      ratifying_provider_can_make_decisions: make_d2,
+    )
+  end
```

Comment:
Can this be moved to a `let` an setup the required data within a context wrapping the example isntead.

Reply:
We only use this when it's needed and when we use it we need full control of which side has `make_decisions`.